libvast/src/system/passive_partition.cpp

```diff
@@ -412,12 +433,7 @@ partition_actor::behavior_type passive_partition(
           self->quit();
           return;
         }
-        if (self->state.id != id) {
-          VAST_ERROR("unexpected ID for passive partition: expected {}, got {}",
-                     id, self->state.id);
-          self->quit();
-          return;
-        }
+        VAST_ASSERT(self->state.id == id);
```

Comment:
Sorry, I meant to put the assertion at the check below.

I don't think we can assert here, because `self->state.id` will contain whatever we found on the filesystem (which is essentially user input) and `id` is what we expected to find, so a mismatch shouldn't trigger an assertion failure.